src/PolyType.Examples/CborSerializer/CborSerializer.cs

```diff
@@ -48,6 +48,20 @@ public static CborConverter<T> CreateConverter<T>(ITypeShapeProvider shapeProvid
     public static CborConverter<T> CreateConverterUsingReflection<T>() =>
         CreateConverter<T>(ReflectionProvider.ReflectionTypeShapeProvider.Default);
 
+    /// <summary>
+    /// Builds an <see cref="CborConverter{T}"/> instance from the specified shape.
+    /// </summary>
+    /// <typeparam name="T">The type for which to build the converter.</typeparam>
+    /// <returns>An <see cref="CborConverter{T}"/> instance.</returns>
+    /// <exception cref="NotSupportedException">No source generated implementation for <typeparamref name="T"/> was found.</exception>
+    public static CborConverter<T> CreateConverterUsingSourceGen<
```

Comment:
@AArnott because of how generic type constraints work, this forces the introduction of separate method groups for source gen and reflection. I don't think this is necessarily a problem if design carefully.